v0/destinations/snapchat_custom_audience/transform.js

```diff
@@ -0,0 +1,226 @@
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json"
+  };
+
+  return response;
+};
+
+const removeUserResponseBuilder = (metadata, message, { Config }) => {
+  const { schema, audienceId } = Config;
+
+  const payload = { id: `${audienceId}`, schema: [], data: [] };
+  let schemaType;
+  let hashedProperty;
+  switch (schema) {
+    case "email":
+      schemaType = "EMAIL_SHA256";
+      hashedProperty = message.properties.listData.add.email;
+      hashedProperty = sha256(hashedProperty.toLowerCase().trim());
+      break;
+    case "phone":
+      schemaType = "PHONE_SHA256";
+      hashedProperty =
+        message.properties.listData.add.phone ||
+        message.properties.listData.add.mobile;
+      hashedProperty = sha256(
+        hashedProperty
+          .toLowerCase()
+          .trim()
+          .replace(/^0+/, "")
+          .replace(/[^0-9]/g, "")
+      );
+      break;
+    case "mobileAdId":
+      schemaType = "MOBILE_AD_ID_SHA256";
+      hashedProperty =
+        message.properties.listData.add.mobileId ||
+        message.properties.listData.add.mobileAdId ||
+        message.properties.listData.add.mobile_id;
+      hashedProperty = sha256(hashedProperty.toLowerCase());
+      break;
+    default:
+      throw new CustomError("Invalid schema", 400);
+  }
+
+  payload.schema.push(schemaType);
+  payload.data.push(hashedProperty);
+
+  const response = {
+    version: "1",
+    type: "REST",
+    method: "DELETE",
+    endpoint: "",
+    headers: {},
+    params: {},
+    body: {
+      JSON: {},
+      JSON_ARRAY: {},
+      XML: {},
+      FORM: {}
+    },
+    files: {}
+  };
+
+  response.endpoint = `${BASE_URL}/segments/${audienceId}/users`;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  const accessToken = getAccessToken(metadata);
+  //   response.params = { listId: Config.listId, customerId: filteredCustomerId };
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json"
+  };
+
+  return response;
+};
+
+const processEvent = async (metadata, message, destination) => {
+  const response = [];
+  if (!message.type) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message Type is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message properties is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties.listData) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::listData is not present inside properties. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (message.type.toLowerCase() === "audiencelist") {
+    let payload;
+    if (message.properties.listData.add) {
```

Comment:
What if the payload contains both add and remove?

Reply:
Considering both we are generating 2 responses

Reply:
That is fine to have two responses.